### PR TITLE
#68 Redundant parentheses should not be used

### LINKED ISSUE #68: Rule: Redundant parentheses should not be used
Replaces http://jira.sonarsource.com/browse/LINTVS-177

In C# when you call a constructor, you can specify property values as well for the initialized object:
`new SomeClass(1,2,3) \{ Property1 = 5, Property2 = 15\}`

But if there is no constructor argument this looks like the following:
`new SomeClass() \{ Property1 = 5, Property2 = 15\}`, where `()` is not needed so could be simplified to `new SomeClass \{ Property1 = 5, Property2 = 15\}`

R# ID: *RedundantEmptyObjectCreationArgumentList*

Implements http://jira.sonarsource.com/browse/RSPEC-3235

### PR DESCRIPTION
**Pull request by tvsonar**
Fixes #68

### CODE REVIEW

**Comment by tvsonar** [comment]
@dbolkensteyn This rule reports loads of issues on generated code. So we should really invest into #85, and then check if the situation got better here or not. Maybe we should exclude this rule from the default profile.

**Comment by tvsonar** [comment]
@ganncamp Please make sure that when you validate this PR, you also check the new error locations of S1854, and S2996.

**Comment by dbolkensteyn** [comment]
@tvsonar It is a bad idea to mix several pull requests. This one should only have been about the new rule, and another one (with an associated GitHub issue) should be there to update the behavior of the other rules

File: src/SonarLint/Rules/RedundantParentheses.cs [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+    [SqaleSubCharacteristic(SqaleSubCharacteristic.Readability)]
+    [Rule(DiagnosticId, RuleSeverity, Title, IsActivatedByDefault)]
+    [Tags("unused")]
+    public class RedundantParentheses : DiagnosticAnalyzer
+    {
+        internal const string DiagnosticId = "S3235";
+        internal const string Title = "Redundant parentheses should not be used";
+        internal const string Description =
+            "Redundant parentheses are simply wasted keystrokes, and should be removed.";
+        internal const string MessageFormat = "Remove the redundant parentheses from this call.";
```
Comment (dbolkensteyn) [inline src/SonarLint/Rules/RedundantParentheses.cs]: @tvsonar I've updated the message into: `Remove these redundant parentheses.`

Indeed, there is no method call in the places where redundant parentheses are reported by this method.

**Pull request merged**